File: src/main/java/com/corrodinggames/rts/gameFramework/graphics/GraphicsEngine.java
package com.corrodinggames.rts.gameFramework.graphics;

import android.content.Context;
import android.graphics.Paint;
import android.graphics.PorterDuff;
import android.graphics.Rect;
import android.graphics.RectF;
import java.io.File;
import java.io.InputStream;
import java.util.concurrent.locks.Lock;

/* JADX INFO: renamed from: com.corrodinggames.rts.gameFramework.m.y */
/* JADX INFO: loaded from: game-lib.jar:com/corrodinggames/rts/gameFramework/m/y.class */
public interface GraphicsEngine {
    GraphicsEngine b(Texture texture);

    GraphicsEngine a(Texture texture);

    boolean a();

    void a(Context context);

    void b();

    GraphicsInterface d();

    void a(GraphicsInterface graphicsInterface);

    void a(AudioRenderer audioRenderer);

    Texture a(int i);

    Texture a(int i, boolean z);

    Texture a(InputStream inputStream, boolean z);

    Texture a(int i, int i2, boolean z);

    Texture b(int i, int i2, boolean z);

    void e();

    void a(Texture texture, float f, float f2, float f3, Paint paint);

    void a(Texture texture, Rect rect, float f, float f2, float f3, Paint paint);

    void a(Texture texture, Rect rect, Rect rect2, Paint paint);

    void a(Texture texture, Rect rect, RectF rectF, Paint paint);

    void a(Texture texture, float f, float f2, Paint paint);

    void a(Texture texture, float f, float f2, Paint paint, float f3, float f4);

    void b(Texture texture, float f, float f2, Paint paint);

    void b(Texture texture, Rect rect, Rect rect2, Paint paint);

    void a(Rect rect, Paint paint);

    void a(boolean z);

    void f();

    void a(Texture texture, Rect rect, Paint paint);

    void a(Texture texture, Rect rect, Paint paint, int i, int i2, int i3, int i4);

    void a(Texture texture, RectF rectF, Paint paint, float f, float f2, int i, int i2);

    void b(int i);

    void a(int i, PorterDuff.Mode mode);

    void a(String str, float f, float f2, Paint paint, Paint paint2, float f3);

    void a(String str, float f, float f2, Paint paint);

    void b(Rect rect, Paint paint);

    void a(RectF rectF, Paint paint);

    void g();

    void h();

    void c(Rect rect, Paint paint);

    void a(Rect rect);

    void a(RectF rectF);

    void a(float f, float f2, float f3, Paint paint);

    void b(float f, float f2, float f3, Paint paint);

    void a(float[] fArr, int i, int i2, Paint paint);

    void i();

    void j();

    void k();

    void l();

    void a(float f, float f2, float f3);

    void a(float f, float f2);

    void a(float f, float f2, float f3, float f4);

    void b(float f, float f2);

    void a(GraphicsOperation graphicsOperation);

    void a(float f, float f2, float f3, float f4, Paint paint);

    int m();

    int n();

    void a(int i, int i2);

    void o();

    void a(Paint paint);

    void p();

    void q();

    int a(String str, Paint paint);

    int b(String str, Paint paint);

    Texture r();

    void a(Texture texture, File file);

    void a(Lock lock);

    void b(Lock lock);

    void a(ShaderProgram shaderProgram);

    float s();
}

File: core/src/main/kotlin/com/corrodinggames/rts/gameFramework/graphics/GraphicsEngine.kt
package com.corrodinggames.rts.gameFramework.graphics

import io.github.rwx.geometry.Rect
import io.github.rwx.geometry.RectF
import io.github.rwx.render.canvas.KoolCanvasBlendMode
import io.github.rwx.render.canvas.KoolPaint
import java.io.File
import java.io.InputStream
import java.util.concurrent.locks.Lock

enum class RenderTargetMode {
    DEFAULT,
    IMMEDIATE,
}

data class GraphicsBackendCapabilities(
    val fixedLayerBufferPixelSize: Int = 0,
    val extraLayerBufferCells: Int = 0,
    val layerBufferScrollPreloadWorldMargin: Int = 0,
    val clearLayerBuffersBeforeCopy: Boolean = false,
    val supportsLayerBufferPreRendering: Boolean = true,
    val supportsSmoothFogLayerBuffers: Boolean = true,
    val requiresFogAtlasLock: Boolean = false,
    val requiresImageTintColorFilter: Boolean = false,
)

fun interface DrawTimeOperation {
    fun draw(graphicsEngine: GraphicsEngine)
}

interface GraphicsEngine {
    fun backendCapabilities(): GraphicsBackendCapabilities = GraphicsBackendCapabilities()

    fun supportsShaderEffects(): Boolean = false

    fun supportsPostProcessing(): Boolean = supportsShaderEffects()

    fun supportsTeamShaders(): Boolean = supportsShaderEffects()

    fun b(texture: Texture?): GraphicsEngine

    fun b(texture: Texture?, mode: RenderTargetMode): GraphicsEngine = b(texture)

    fun a(lock: Lock?)

    fun b(lock: Lock?)

    fun a(operation: DrawTimeOperation?) {
        operation?.draw(this)
    }

    fun a(i: Int): Texture

    fun a(i: Int, z: Boolean): Texture

    fun a(inputStream: InputStream?, z: Boolean): Texture

    fun a(i: Int, i2: Int, z: Boolean): Texture

    fun b(i: Int, i2: Int, z: Boolean): Texture

    fun a(texture: Texture?, f: Float, f2: Float, f3: Float, paint: KoolPaint?)

    fun a(texture: Texture?, rect: Rect?, f: Float, f2: Float, f3: Float, paint: KoolPaint?)

    fun a(texture: Texture?, rect: Rect?, rect2: Rect?, paint: KoolPaint?)

    fun a(texture: Texture?, rect: Rect?, rectF: RectF?, paint: KoolPaint?)

    fun a(texture: Texture?, f: Float, f2: Float, paint: KoolPaint?)

    fun a(texture: Texture?, f: Float, f2: Float, paint: KoolPaint?, f3: Float, f4: Float)

    fun b(texture: Texture?, f: Float, f2: Float, paint: KoolPaint?)

    fun b(texture: Texture?, rect: Rect?, rect2: Rect?, paint: KoolPaint?)

    fun a(rect: Rect?, paint: KoolPaint?)

    fun a(texture: Texture?, rect: Rect?, paint: KoolPaint?)

    fun a(texture: Texture?, rect: Rect?, paint: KoolPaint?, i: Int, i2: Int, i3: Int, i4: Int)

    fun a(texture: Texture?, rectF: RectF?, paint: KoolPaint?, f: Float, f2: Float, i: Int, i2: Int)

    fun b(i: Int)

    fun a(i: Int, mode: KoolCanvasBlendMode?)

    fun a(str: String?, f: Float, f2: Float, paint: KoolPaint?, paint2: KoolPaint?, f3: Float)

    fun a(str: String?, f: Float, f2: Float, paint: KoolPaint?)

    fun b(rect: Rect?, paint: KoolPaint?)

    fun a(z: Boolean)

    fun f()

    fun a(rectF: RectF?, paint: KoolPaint?)

    fun c(rect: Rect?, paint: KoolPaint?)

    fun a(rect: Rect?)

    fun a(rectF: RectF?)

    fun a(f: Float, f2: Float, f3: Float, paint: KoolPaint?)

    fun b(f: Float, f2: Float, f3: Float, paint: KoolPaint?)

    fun a(fArr: FloatArray?, i: Int, i2: Int, paint: KoolPaint?)

    fun i()

    fun j()

    fun k()

    fun l()

    fun a(f: Float, f2: Float, f3: Float)

    fun a(f: Float, f2: Float)

    fun a(f: Float, f2: Float, f3: Float, f4: Float)

    fun b(f: Float, f2: Float)

    fun a(f: Float, f2: Float, f3: Float, f4: Float, paint: KoolPaint?)

    fun m(): Int

    fun n(): Int

    fun a(i: Int, i2: Int)

    fun o()

    fun p()

    fun q()

    fun a(shaderProgram: ShaderProgram?)

    fun a(str: String?, paint: KoolPaint?): Int

    fun b(str: String?, paint: KoolPaint?): Int

    fun r(): Texture

    fun a(texture: Texture?, file: File?)
}
